File: proxy/python/mask_order.py
from orderable_mask import OrderableMask
from proxy.python.person import Person


class MaskOrder(OrderableMask):
    """
    Maske siparişininin oluşturulduğu sınıf.
    İstemcinin direkt erişmesini istemediğimiz alandır.
    UML diyagramındaki ConcreteService sınıfına denk gelmektedir.
    """

    def create_order(self, person: Person):
        # Burada siparişin oluşturulma işlemleri yer alacaktır.
        print(f"Order created to {person.address}.")

File: proxy/go/mask-order.go
package main

import "fmt"

// Maske siparişininin oluşturulduğu sınıf.
// İstemcinin direkt erişmesini istemediğimiz alandır.
// UML diyagramındaki ConcreteService sınıfına denk gelmektedir.
type maskOrder struct{}

func NewMaskOrder() *maskOrder {
	return &maskOrder{}
}

func (o maskOrder) CreateOrder(p Person) {
	// Burada siparişin oluşturulma işlemleri yer alacaktır.
	fmt.Println("Order created to " + p.address)
}
